runtime-modules/membership/src/lib.rs

```diff
@@ -1,9 +1,647 @@
+                <MemberIdsByRootAccountId<T>>::mutate(&membership.root_account, |ids| {
+                    ids.retain(|id| *id != member_id);
+                });
+
+                <MemberIdsByRootAccountId<T>>::mutate(&new_root_account, |ids| {
+                    ids.push(member_id);
+                });
+
+                membership.root_account = new_root_account.clone();
+                Self::deposit_event(RawEvent::MemberSetRootAccount(member_id, new_root_account));
+            }
+        }
+
+        pub fn add_screened_member(
+            origin,
+            new_member_account: T::AccountId,
+            handle: Option<Vec<u8>>,
+            avatar_uri: Option<Vec<u8>>,
+            about: Option<Vec<u8>>
+        ) {
+            // ensure sender is screening authority
+            let sender = ensure_signed(origin)?;
+
+            if <ScreeningAuthority<T>>::exists() {
+                ensure!(sender == Self::screening_authority(), "not screener");
+            } else {
+                // no screening authority defined. Cannot accept this request
+                return Err("no screening authority defined");
+            }
+
+            // make sure we are accepting new memberships
+            ensure!(Self::new_memberships_allowed(), "new members not allowed");
+
+            let user_info = Self::check_user_registration_info(handle, avatar_uri, about)?;
+
+            // ensure handle is not already registered
+            Self::ensure_unique_handle(&user_info.handle)?;
+
+            let member_id = Self::insert_member(&new_member_account, &user_info, EntryMethod::Screening(sender));
+
+            Self::deposit_event(RawEvent::MemberRegistered(member_id, new_member_account));
+        }
+
+        pub fn set_screening_authority(origin, authority: T::AccountId) {
+            ensure_root(origin)?;
+            <ScreeningAuthority<T>>::put(authority);
+        }
+    }
+}
+
+/// Reason why a given member id does not have a given account as the controller account.
+pub enum ControllerAccountForMemberCheckFailed {
+    NotMember,
+    NotControllerAccount,
+}
+
+pub enum MemberControllerAccountDidNotSign {
+    UnsignedOrigin,
+    MemberIdInvalid,
+    SignerControllerAccountMismatch,
+}
+
+pub enum MemberControllerAccountMismatch {
+    MemberIdInvalid,
+    SignerControllerAccountMismatch,
+}
+pub enum MemberRootAccountMismatch {
+    MemberIdInvalid,
+    SignerRootAccountMismatch,
+}
+
+impl<T: Trait> Module<T> {
+    /// Provided that the member_id exists return its membership. Returns error otherwise.
+    pub fn ensure_membership(id: T::MemberId) -> Result<MembershipOf<T>, &'static str> {
+        if <MembershipById<T>>::exists(&id) {
+            Ok(Self::membership(&id))
+        } else {
+            Err("member profile not found")
+        }
+    }
+
+    /// Ensure that given member has given account as the controller account
+    pub fn ensure_is_controller_account_for_member(
```

Comment:
I think this is exclusively used for authenticating claimed members, both in this module, and by other caller module. We may as well make it `ensure_member_signed_as_controller(origin, member_id)`.

Reply:
Yes. And when I try to create it, I found a similar method with a slightly different name ensure_member_controller_account_signed() nearby.

Reply:
Ok, so we remove this then?

Reply:
Removed that entry, but found another one more complicated use-case in the content working group in the apply_on_curator_opening(). So we keep it for now.